### ISSUE TITLE
Safari "A problem repeatedly occured on domain.x" on iOS

### ISSUE THREAD

**Issue by efsjouw** [google/model-viewer#1200]
### Description

I only get this error in Safari on iOS when there are a lot of model-viewers on the page. Even if content is only lazy-loaded on interaction.

#### Live Demo
https://recreate-glb-samples.glitch.me/ 

### Browser Affected

- [ ] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Helios
- [ ] IE
- [x] Safari

### OS

- [ ] Android
- [x] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### Versions

I use there sources
https://unpkg.com/@google/model-viewer/dist/model-viewer.js
https://unpkg.com/@google/model-viewer/dist/model-viewer-legacy.js

**Comment by elalish** [comment]
@efsjouw It seems to be working fine on my iPhone SE. Can you give more specific repro steps? If the problem is only when you've loaded a bunch of the models, it's probably just running out of GPU ram, which there's not much we can do about. You can either compress your models (especially trim down texture sizes) or set up your page to swap `src` instead of having them all on the page at once. Then our modelCache takes care of switching and clearing out old models automatically.

**Comment by efsjouw** [comment]
It is a 6s Plus, I added some load model buttons now so it wouldn't start loading all the models at once but still appears to have some problems.

**Comment by phreak** [comment]
I experience the same and actually would like to deactivate loading at all and only use the device recognition (Android, iOS) on mobile devices to call the AR functionality.

**Comment by elalish** [comment]
@phreak Can you tell me more about your use case? What happens for browsers that don't support AR, as in neither webXR nor native apps?

**Comment by phreak** [comment]
@elalish: Thank you for asking.
Basically I show about 7 models on a page which works fine on a Mac Desktop (Chrome) and and a recent iPad Pro (Safari). One model is on the top of the page and 6 are below the fold. Loading the page on mobile with iPhone Xs Safari appears to load the page first regulary but then within a few seconds a second time (due to technical issues I guess). The page works as long as I do not get the 6 models into the viewport of the mobile device. Then it errors out with "A problem repeatedly occurred on URL". While the 6 models are set to loading="lazy" and reveal="interaction", they still start to load when in the viewport. But it would be totally fine and probably also helpful to only show the poster image and AR functionalities on mobile.

Regarding devices that do not support AR. In my case that would be fine. It's an AR centered site and people would only go there if they are interested in the topic, though an alert (You need an AR enabled device to see this) for these cases would be great.

**Comment by elalish** [comment]
@phreak It sounds like a bug if you've set loading="lazy" and reveal="interaction" and the models are still loading when they come in the viewport. Could you file a separate issue for that with a glitch to demonstrate? I'd like to get that fixed, which may solve this problem as well, at least partially.

**Comment by 60days** [comment]
I'd like to report also encountering this with a single object/model viewer element on page, across iPhone 6s, 7, X and 11 devices. On the devices with more RAM (X/11)there does need to be a few other tabs open in the browser (not many though, just four or five).

Since the crash disconnects any profilers/dev tools I can't report anything more than the error message. I have 'fixed' the issue on my production sites by rolling back to 1.3.0, so I believe the specific issue was introduced after that release.

This model replicates the issue using the latest release version on the CDN: https://www.kordainteriors.com.au/crash/index.html?callToAction=SHOP&checkoutTitle=Carve+Stone&checkoutSubtitle=New+Zealand+wool+and+Cotton+Viscose+2x1.5m&productCode=HTNZ001STONESMALL&ctaUrl=https://www.kordadesign.com.au/product-page/carve-2 it has 3x 4K textures which is heavy for a web-based model, but shouldn't be hitting memory limits in the active app afaik. 

It may be replicatable without crashes by using less heavy models and profiling memory-differences between release versions of model-viewer.

**Comment by paella55** [comment]
I also had this issue and in my case it was due to using a very large `skybox-image`. It's resolution was 10.000 x 5.000, which is not a power of 2 and also it is extremely large. I had to reduce it to 1024 x 516.

**Comment by Michele-Telesca** [comment]
i am having the same problem on some ios devices. The models have been optimized KTX2. Can anyone tell me some possible solution? The error appears independently of the Broswer on iphone Xs and sometimes on IPhone 11.
 [](https://ar3d.spaarkly.it/custom-view/Maccaturo/SQOPgnVDVW66skxfFW4O )
  [](https://ar3d.spaarkly.it/custom-view/Maccaturo/TBpxVaN5iSzZv17YzMLp)

**Comment by elalish** [comment]
I don't believe we support USDZ autogeneration for KTX2 textures yet. I think you'll have to convert them with a  different tool and set an `ios-src` in addition.
